client/pages/auth/index.tsx

```diff
@@ -26,6 +27,24 @@ function ConnectButton() {
   };
   const Button = isConnected() ? ButtonGreen : ButtonGray;
 
+  useEffect(() => {
+    if (account) {
+      fetch('/api/auth', {
+        method: 'POST',
+        body: JSON.stringify({ walletAddress: account }),
+      })
+        .then((response) => {
+          console.log(response);
+          return response.json();
+        })
+        .then((result) => {
+          console.log(result);
+          // Now we can set the user in a global auth context and redirect them
+          // to /create-profile if they don't have one yet
+        });
+    }
+  }, [account]);
```

Comment:
We could follow the `async`/`await` pattern we're using elsewhere if we moved logic into a closure

```suggestion
  useEffect(() => {
  	const fetchAccount = async () => {
  	  try {
  	    const response = await fetch('/api/auth', {
	        method: 'POST',
  	      body: JSON.stringify({ walletAddress: account }),
      	})
      	console.log(response)

      	const account = response.json()
      	console.log(account)
      	
      	// Now we can set the user in a global auth context and redirect them
        // to /create-profile if they don't have one yet
  	  } catch (error) {
  	    console.error(error)
  	  }
  	}

    if (account) {
      fetchAccount()
    }
  }, [account]);
```

I suppose that might complicate error handling a bit if we need to do separate checks after each call, but I prefer trying to keep consistency around `async` stuff.

One benefit to this route is that we could promote the callback to a named `export`  fn with an `account` parameter, which would let us write tests for the different paths this fn could follow.